### PR TITLE
feat: track per account storage usage of media

### PR DESCRIPTION
**Pull request by blushi**
## Description

Closes: #421

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [x] targeted `dev` branch
- [x] provided a link to the relevant issue or specification
- [x] reviewed "Files changed" and left comments if necessary
- [x] confirmed all CI checks have passed
- [ ] submit a PR against `master` branch after this one (and other PRs depending on this one, e.g. on regen-network/regen-web, if applicable) get merged

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please **add
your handle next to the items reviewed if you only reviewed selected items**.*

I have...

- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed code correctness and readability
- [ ] reviewed documentation is accurate
- [ ] reviewed tests
- [ ] manually tested (if applicable)

### CODE REVIEW

File: migrations/current.sql [hunk 1]
```diff
@@ -1 +1,19 @@
--- Enter migration here
+DROP TABLE IF EXISTS upload;
+CREATE TABLE upload (
+  id uuid DEFAULT public.uuid_generate_v1() NOT NULL,
+  created_at timestamp with time zone DEFAULT now(),
+  url text UNIQUE NOT NULL,
+  size int NOT NULL,
```
Comment (blushi) [inline migrations/current.sql:6]: this is the size in bytes
maybe we could add mimetype too

File: server/routes/files.ts [hunk 2]
```diff
@@ -10,28 +10,24 @@ import {
...
 router.post(
   '/files',
   async (request: UserRequest, response: express.Response, next) => {
-    let client: undefined | PoolClient = undefined;
+    let client: undefined | PoolClient = undefined,
+      projectId: undefined | string;
+    const currentAccountId = request.user?.accountId;
     try {
-      const image = (request as FilesRequest).files.image;
+      const image = request.files?.image as UploadedFile;
```
Comment (blushi) [inline server/routes/files.ts:30]: this endpoint was primarily made for uploading images but we could update it to be more generic just by turning the request `image` into a `file` without any further changes

File: server/routes/files.ts [hunk 3]
```diff
@@ -82,12 +80,29 @@ router.post(
...
       } else {
+        const url = getFileUrl({
+          bucketName,
+          key,
+          fileName: image.name,
+        });
+
+        // Track file storage for projects
+        if (projectId && client) {
+          const extension = image.name.split('.').pop();
```
[the remark this answers is not in the document]
Comment (blushi) [reply server/routes/files.ts:96]: I've updated this based on the list in https://github.com/regen-network/regen-ledger/pull/2102/files#diff-fde0c4951628ea8b873d1d60ba4807e4c1cae9e9c079bbc66f6eeae88ae217f1R86-R87

[the remark this answers is not in the document]
Comment (blushi) [reply server/routes/files.ts:96]: added a check, falling back to `bin` extension for now

File: migrations/committed/000046.sql [hunk 4]
```diff
@@ -0,0 +1,23 @@
+--! Previous: sha1:a38c8e478d6d4efa0a0c01805cd4030c059f241a
+--! Hash: sha1:59c7d3d8f53b052345bbacafba5d712e77f22e58
+
+DROP TABLE IF EXISTS upload;
+CREATE TABLE upload (
```
Comment (blushi) [inline migrations/committed/000046.sql:5]: I haven't set up any `GRANT`s on this table to our `app_user` (marketplace app user) on purpose. For now, only the root user has access to it.

**Review by aaronc — APPROVED** [review]

**Pull request merged**